crates/ruma-state-res/src/event_auth.rs

```diff
@@ -446,36 +472,22 @@ fn valid_membership_change(
         join_rules = from_json_str::<RoomJoinRulesEventContent>(jr.content().get())?.join_rule;
     }
 
-    if let Some(prev) = prev_event {
-        if *prev.event_type() == EventType::RoomCreate && prev.prev_events().next().is_none() {
-            return Ok(true);
-        }
-    }
-
     let power_levels_event_id = power_levels_event.as_ref().map(|e| e.event_id());
     let sender_membership_event_id = sender_membership_event.as_ref().map(|e| e.event_id());
     let target_user_membership_event_id =
         target_user_membership_event.as_ref().map(|e| e.event_id());
 
-    // FIXME: `JoinRule::Restricted(_)` can contain conditions that allow a user to join if
-    // they are met. So far the spec talks about roomId based auth inheritance, the problem with
-    // this is that ruma-state-res can only request events from one room at a time :(
```

Comment:
I think this would be too hard to do in ruma and I will just do it in conduit for now. I can still restore the comment if you want

Reply:
Reading this again, I think the comment was wrong. We will never check the other rooms ourselves when we validate an incoming event. We trust the authorized_user_server. We only look at the other room if we create a new join event in Conduit.

Reply:
The server spec says nothing about the room inheritance stuff but if you look here: 
https://spec.matrix.org/v1.2/client-server-api/#restricted-rooms
specifically
https://spec.matrix.org/v1.2/client-server-api/#conditions

I don't see another way of implementing that other than on the server allowing the event if the user is in the specified room?

FYI for our purposes the join rule would look like this `JoinRule::Restricted(Restricted { allow: vec![AllowRule::RoomMembership(RoomMembership { room_id })] })`

Reply:
This is not verified over federation, only the origin server checks this

Reply:
Why does that matter? Don't you still use this function (indirectly, through `auth_check`) to validate events before their sent and for the membership event route?

I may be missing where the logic of this would be added if it's removed from state-res, that might be my confusion. I could imagine homeservers having to special case this code path so they avoid what `state_res::auth_check` does when it's a member event and the join rule is restricted but if that's the case that's what the comment is for. maybe it would be more helpful as an invariant documented as part of the functions doc comments (auth_check and valid_membership_change).

Reply:
used here https://gitlab.com/famedly/conduit/-/blob/next/src/database/rooms.rs#L1888